src/libcore/marker.rs

```diff
@@ -356,8 +356,48 @@ pub trait PhantomFn<A:?Sized,R:?Sized=()> { }
 ///
 /// # Examples
 ///
-/// When handling external resources over a foreign function interface, `PhantomData<T>` can
-/// prevent mismatches by enforcing types in the method implementations:
+/// ## Unused lifetime parameter
+///
+/// Perhaps the most common time that `PhantomData` is required is
+/// with a struct that has an unused lifetime parameter, typically as
+/// part of some unsafe code. For example, here is a struct `Slice`
+/// that has two pointers of type `*mut T`, presumably pointing into
+/// an array somewhere:
+///
+/// ```
+/// struct Slice<'a, T> {
+///     start: *mut T,
+///     end: *mut T,
+/// }
+/// ```
+///
+/// The intention is that the underlying data is only valid for the
+/// lifetime `'a`, so `Slice` should not outlive `'a`. However, this
+/// intent is not expressed in the code, since there are no uses of
+/// the lifetime `'a` and hence it is not clear what data it applies
+/// to. We can correct this by telling the compiler to act *as if* the
+/// `Slice` struct contained a borrowed reference `&'a T`:
+///
+/// ```
+/// # use std::marker::PhantomData;
```

Comment:
I think we've been showing the `use` statements more lately, as people try to copy/paste examples and then wonder why it doesn't compile